lib/index.js

```diff
@@ -1,11 +1,19 @@
 'use strict';
 
+const RateLimiter = require('./rate-limiter');
+
 exports.register = (server, options, next) => {
 
   server.ext('onPostAuth', (request, reply) => {
-    request.payload.loaded = true;
+    return RateLimiter.limit(request, options.redisClient, options.defaultRate)
+    .then((rate) => {
+      if (rate && rate.current > rate.limit) {
+        return reply(options.overLimitHandler());
+      }
 
-    return reply.continue();
+      request.payload.rate = rate;
```

Comment:
I don't think you can alter the payload object as that can affect requests. What information do you need to be passing on here in the request object?